public/react-services/wz-agents.ts

```diff
@@ -30,3 +34,16 @@ export function hasAgentSupportModule(agent, component){
   const agentOSType = getAgentOSType(agent);
   return !(UnsupportedComponents[agentOSType].includes(component));
 };
+
+
+export async function getAuthorizedAgents() {
+  const agentsList: IApiResponse<{id: string}> = await WzRequest.apiReq('GET', `/agents`, {})
+  .catch(error => {   
+    getToasts().addError(error, {title: `Error getting user authorized agents`} as ErrorToastOptions);
+    return Promise.reject();
+  });
+
+  const allowedAgents = agentsList ? agentsList.data.data.affected_items.map((agent) => agent.id) : []
+
+  return allowedAgents;
+}
```

Comment:
I am not sure why mix `async/await` with `.catch`

Reply:
Okay, I have changed .catch to a try catch statement. Thanks for reporting it!